Month2/week1/Month2-Project Week 1-Remembering the Bases/VideoGame/CategoryContainer.cpp

```diff
@@ -0,0 +1,43 @@
+#include "CategoryContainer.h"
+FCategoryContainer::FCategoryContainer()
+{
+	CategoriesCount = 0;
+}
+int FCategoryContainer::GetCategoryCount()const 
+{
+	return CategoriesCount;
+}
+void FCategoryContainer::AddCategory(FCategory NewCategory)
+{
+	Categories[CategoriesCount] = NewCategory;
+	CategoriesCount += 1;
+}
+bool FCategoryContainer::IsValidIndex(int Index)
+{
+	if (!std::cin.good() || Index > 4 || Index < 0)return false;
```

Comment:
An index has nothing to do with the std::cin state. An index is just an index, so to check for its validity, you should be checking if it is between 0 and 4 and that's it.

After that, this code could be collapsed into a single line - that if-else is redundant.

You could just do:
```
return Index >= 0 && Index <= 4;
```